src/Riok.Mapperly/Symbols/SetterMemberPath.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Riok.Mapperly.Descriptors;
+
+namespace Riok.Mapperly.Symbols;
+
+public class SetterMemberPath : MemberPath
+{
+    private SetterMemberPath(IReadOnlyList<IMappableMember> path, bool isMethod)
+        : base(path)
+    {
+        IsMethod = isMethod;
+    }
+
+    public bool IsMethod { get; }
```

Comment:
> I had a super hard time understanding what exactly this does. IMO this makes things really complicated. This seems to be related to setting fields via the ref return. This could get a lot simpler without this parameter, by setting fields via the accessor too.

`IsMethod` indicates that the setter must be invoked as it is an `UnsafeAccessor` for a property set. It has nothing to do with setting fields via `ref` return as the `UnsafeAccessor` for fields have the exact same usage and semantics as a normal property or field. `IsMethod` is only true for setters for `UnsafeAccessor`.

>  Is there a special reason for not doing this? (same for MethodMember).

At one point I did try a solution similar to what you suggested. Unfortunately it didn't work for `MemberNullAssignmentInitializerMapping` (`source.Value ??= new()`). The caller has to know if the `SetterMemberPath` is a method or not because you have to provide it with different sources. ie `new()` for a normal field, property or `UnsafeAccessor` field, whereas `source.GetValue() ?? new()` must be provided for an `UnsafeAccessor` for setters. See `MethodMemberNullAssignmentInitializerMapping`. I ended up with my current solution to handle these edge cases.

I'll try your suggestion with the `coalesceAssignment` parameter. Unfortunately I had to keep `IsMethod`, I might be missing something.

Reply:
Got it now 😊  Looks good to me.